examples/download_progress/src/main.rs

```diff
@@ -1,144 +1,277 @@
+    }
+}
+
+impl Task {
+    fn view(&mut self) -> Element<Message> {
+        let download_text = |text| {
+            Text::new(text)
+                .size(15)
+                .horizontal_alignment(HorizontalAlignment::Center)
+        };
+        Row::new()
+            .spacing(20)
+            .align_items(Align::Center)
+            .push(Text::new(&self.name))
+            .push(match self.state {
+                TaskState::Downloading(progress) => {
+                    ProgressBar::new(0.0..=100.0, progress)
+                        .width(Length::Units(150))
+                        .height(Length::Units(18))
+                        .into()
+                }
+                _ => {
+                    let element: Element<_> =
+                        Space::new(Length::Units(150), Length::Shrink).into();
+                    element
+                }
+            })
+            .push(
+                match self.state {
+                    TaskState::Idle => {
+                        Button::new(&mut self.button, download_text("Download"))
+                            .on_press(Message::StartDownload(
+                                (*self.url).to_string(),
+                            ))
+                    }
+                    TaskState::Downloading(progress) => Button::new(
+                        &mut self.button,
+                        download_text(&format!("{:.2}%", progress)),
+                    )
+                    .on_press(Message::CancelDownload((*self.url).to_string())),
+                    TaskState::Finished => Button::new(
+                        &mut self.button,
+                        download_text("Downloaded"),
+                    ),
+                    TaskState::Error => {
+                        Button::new(&mut self.button, download_text("Errored"))
+                            .on_press(Message::StartDownload(
+                                (*self.url).to_string(),
+                            ))
+                    }
+                }
+                .width(Length::Units(85)),
+            )
+            .into()
+    }
+}
+
+impl Example {
+    async fn load_data() -> State {
+        let tasks: Vec<_> = (1..=10)
+            .map(|n| Task {
+                name: format!("File {:0>2}", n),
+                url: format!("{}?{}", FILE_URL, n),
+                ..Task::default()
+            })
+            .collect();
+        State {
+            tasks,
+            ..State::default()
+        }
+    }
 }
 
 impl Application for Example {
-    type Executor = executor::Default;
+    type Executor = iced::executor::Default;
     type Message = Message;
     type Flags = ();
 
     fn new(_flags: ()) -> (Example, Command<Message>) {
         (
-            Example::Idle {
-                button: button::State::new(),
-            },
-            Command::none(),
+            Example::Loading,
+            Command::perform(Example::load_data(), Message::Loaded),
```

Comment:
For simplicity, I think we should remove the `Loading` variant and `load_data` altogether.